src/astro/utils/table_handler.py

```diff
@@ -65,13 +65,23 @@ def _set_variables_from_first_table(self):
 
         if first_table:
             self.conn_id = first_table.conn_id or self.conn_id
-            self.database = first_table.database or self.database
-            self.schema = first_table.schema or self.schema
-            self.warehouse = first_table.warehouse or self.warehouse
-            self.role = first_table.role or self.role
+            self.database = (
+                getattr(first_table.metadata, "database", None) or self.database
```

Comment:
This feels odd. Why do we have to use all these getattr?

Reply:
because metadata is optional, while we define table.